gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -169,9 +175,8 @@ public void run() {
         LOG.error("Failed to close all open resources", t);
       }
 
-      long endTime = System.currentTimeMillis();
-      this.taskState.setEndTime(endTime);
-      this.taskState.setTaskDuration(endTime - startTime);
+      this.taskState.setTaskDuration(TimeUnit.NANOSECONDS.toMillis(taskTimerContext.stop()));
+      this.taskState.setEndTime(System.currentTimeMillis());
```

Comment:
Yes, the timer actually goes through the metricRegistry and will be reported along with other metrics. `TaskState` is really the unit of storage in the state store. It keeps some fields capturing the execution info of a task only for informational purpose. If we move to a querable state store, then info stored in TaskState will be useful for query purposes.